src/service.rs

```diff
@@ -44,6 +46,20 @@ pub(super) mod errors {
             }
         }
     }
+
+    #[derive(Debug)]
+    pub enum ProcessGrpcMessageError {
+        Protobuf(ProtobufError),
+        Property(PropertyError),
+        EmptyResponse,
+        UnsupportedField,
+    }
```

Comment:
We don't really do anything with the granularity... If this is a thing to be useful "later", it should be introduced then.
Right now, this could be simplified... and would simplify the `match`es that deals with the error.

Reply:
It turns out that to address this [comment](https://github.com/Kuadrant/wasm-shim/pull/196#discussion_r2209835383), `UnsupportedField` needs to be used ( auth action unsupportedfield errors stop flow regardless of FailureMore). Will remove `EmptyResponse` and added when needed.